pipeline.py

```diff
@@ -72,31 +90,96 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+
+        for article in self.corpus_manager.get_articles().values():
+            article_raw_text = article.get_raw_text()
+
+            cleaned_tokens = []
+            single_tagged_tokens = []
+            multiple_tagged_tokens = []
+
+            for token in self._process(article_raw_text):
+                cleaned_tokens.append(token.get_cleaned())
+                single_tagged_tokens.append(token.get_single_tagged())
+                multiple_tagged_tokens.append(token.get_multiple_tagged())
+
+            article.save_as(' '.join(cleaned_tokens), ArtifactType.cleaned)
+            article.save_as(' '.join(single_tagged_tokens), ArtifactType.single_tagged)
+            article.save_as(' '.join(multiple_tagged_tokens), ArtifactType.multiple_tagged)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        pattern = re.compile(r'[а-яА-Яa-zA-z ё]')
+        for symbol in raw_text:
+            if not pattern.match(symbol):
+                text = raw_text.replace(symbol, '')
+
+        analyzed_text = Mystem().analyze(text)
+        morph = pymorphy2.MorphAnalyzer()
+        tokens = []
+        for token in analyzed_text:
+            if 'analysis' not in token:
```

Comment:
checks should be improved - you finally need to ensure that extracting data like this: `token['analysis'][0]['lex']` from an object that does not contain anything on this path will not crash the whole program

Reply:
done